File: example implementations/java/convertor/src/rifflike/RiffWriter.java
package rifflike;

import static utils.NumberUtilities.intsToBytes;

import java.io.DataOutputStream;
import java.io.IOException;

public class RiffWriter {
    public static void writeNextMagicaChunk(DataOutputStream dos, MagicaChunk chunk) throws IOException {
        dos.writeBytes(chunk.id);
        dos.write(intsToBytes(true, chunk.size));
        dos.write(intsToBytes(true, chunk.subchunkSize));
        dos.write(chunk.content);
        for (MagicaChunk subchunk : chunk.subChunks)
            writeNextMagicaChunk(dos, subchunk);
    }

    public static void writeNextChunk(DataOutputStream dos, Chunk chunk) throws IOException {
        dos.writeBytes(chunk.id);
        dos.write(intsToBytes(true, chunk.size));
        dos.write(chunk.content);
    }
}

File: example implementations/csharp/cvox-convertor/rifflike/RiffWriter.cs
using System.Text;
using static cvox_convertor.utils.NumberUtilities;

namespace cvox_convertor.rifflike
{
    public class RiffWriter
    {
        public static void WriteNextMagicaChunk(Stream stream, MagicaChunk chunk)
        {
            stream.Write(Encoding.ASCII.GetBytes(chunk.Id));
            stream.Write(IntsToBytes(true, chunk.Size));
            stream.Write(IntsToBytes(true, chunk.SubchunkSize));
            stream.Write(chunk.Content);
            foreach (MagicaChunk subchunk in chunk.SubChunks)
                WriteNextMagicaChunk(stream, subchunk);
        }
        public static void WriteNextChunk(Stream stream, Chunk chunk)
        {
            stream.Write(Encoding.ASCII.GetBytes(chunk.Id));
            stream.Write(IntsToBytes(true, chunk.Size));
            stream.Write(chunk.Content);
        }
    }
}
